### ISSUE TITLE
Using the PDF pre-processor

### ISSUE THREAD

**Issue by nevvermind** [tleyden/open-ocr#117]
Firstly, thanks for your effort. It's a cool setup you've done here.

And thanks @xf0e for the [PR](https://github.com/tleyden/open-ocr/pull/108). 

Unfortunately, it seems that one of the Docker images up in DockerHub is out of date. I mean https://hub.docker.com/r/tleyden5iwx/open-ocr-preprocessor. It doesn't seem to contain the PDF pre-processor code added by @xf0e's PR.

```
$ docker images
REPOSITORY                          TAG                 IMAGE ID            CREATED             SIZE
busybox                             latest              d8233ab899d4        4 weeks ago         1.2MB
tleyden5iwx/open-ocr-2              latest              7b3add377eb6        3 months ago        845MB
rabbitmq                            3.6.5-management    5335b737c380        2 years ago         179MB
tleyden5iwx/open-ocr-preprocessor   latest              00a689ddd4f8        4 years ago         1.31GB
```
The `preprocessor_rpc_worker.go` file is different in the built image than the one found in the [master branch](https://github.com/tleyden/open-ocr/blob/1cd43c1659c42dd65487559e9f055436c25b0e06/preprocessor_rpc_worker.go#L31).
```
docker exec docker-compose_strokewidthtransform_1 cat /opt/go/src/github.com/tleyden/open-ocr/preprocessor_rpc_worker.go
```
I'm assuming that's why the `No preprocessor found for: "convert-pdf"` error happens: https://github.com/tleyden/open-ocr/pull/108#issuecomment-468325379

Would you mind checking if the image is up to date with the code, @tleyden, when you've got some time?

**Comment by tleyden** [comment]
> It doesn't seem to contain the PDF pre-processor code added by @xf0e's PR.

Thanks for tracking that down!  I will update this and post to this issue.

**Comment by nevvermind** [comment]
Hi, @tleyden. Any news on the new build?

**Comment by tleyden** [comment]
Hey, sorry I haven't had a chance yet.  I will do so soon.

**Comment by tleyden** [comment]
Ok I kicked off new builds:

https://cloud.docker.com/repository/docker/tleyden5iwx/open-ocr-2/builds
https://cloud.docker.com/repository/docker/tleyden5iwx/open-ocr-preprocessor/builds

(not sure if these are visible to people)

**Comment by nevvermind** [comment]
Thanks, @tleyden.

I can't access the above, but these instead:
- https://hub.docker.com/r/tleyden5iwx/open-ocr-2/builds
- https://hub.docker.com/r/tleyden5iwx/open-ocr-preprocessor/builds

It seems there's no new build for the latter. Should there be?

**Comment by tleyden** [comment]
The build failed with: https://gist.github.com/tleyden/7de181727c5ab843c87654c8054c6843

I haven't looked closely into it yet.  If you have any ideas, please post!

**Comment by nevvermind** [comment]
Looks related to https://github.com/streadway/amqp/issues/291

Do you need to bump the Go version maybe?

**Comment by tleyden** [comment]
Thanks @nevvermind!  Seems likely.

I'm adding a ticket to use [go modules](https://github.com/golang/go/wiki/Modules).

**Comment by nevvermind** [comment]
Hmm, that could take a while, am I right?
I'm wondering how did @xf0e build it without bumping the Go version.

**Comment by xf0e** [comment]
@nevvermind I always try to use the most recently Go version. This is why I never encountered this problem.

**Comment by tleyden** [comment]
@nevvermind yeah, I don't think switching to go modules needs to block this... 

quick fix: bump go version in dockerfile build
deep fix: use go modules so this doesn't keep happening

**Comment by darmanovic** [comment]
Maybe I'm missing the point, I updated go version to 1.12 but problem persists?

**Comment by tleyden** [comment]
@darmanovic you're probably not talking about the go version in the container, which is what matters at build time.

I think the underlying problem is this:

https://github.com/tleyden/docker/blob/master/stroke-width-transform/Dockerfile#L2

`FROM ubuntu:14.04`

It's using an ancient ubuntu, and therefore an ancient Go version.
